### ISSUE TITLE
Authentication fails with entra ID (personal account)

### ISSUE THREAD

**Issue by weikanglim** [Azure/acr#723]
**Describe the bug**
We've received a few reports of users failing to deploy [Aspire](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview) projects using [azd](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/overview) because of auth failures to ACR.

After investigating further, it seems that users logging in with personal accounts to Azure subscriptions are affected. A minimal repro is provided below to use `az` to reproduce the issue.

**To Reproduce**
1. Log in with personal account with an Azure subscription
2. Create an ACR
3. `az acr login --expose-token -n <ACR>`
4. Receive error:
```
Unable to get AAD authorization tokens with message: 2023-12-05 21:52:08.405811 An error occurred: CONNECTIVITY_REFRESH_TOKEN_ERROR
Access to registry '<acr>.azurecr.io' was denied. Response code: 401. Please try running 'az login' again to refresh permissions.
```

**Expected behavior**
Login succeeds, token is printed out.

**Screenshots**

An example response error occuring at 2023-12-05T22:19:27:

```
 POST https://acrrjdeqels63jda.azurecr.io/oauth2/exchange
--------------------------------------------------------------------------------
RESPONSE 401: 401 Unauthorized
ERROR CODE UNAVAILABLE
--------------------------------------------------------------------------------
Response contained no body
--------------------------------------------------------------------------------

x-ms-correlation-request-id: 549b20a24cef3a6f4d60d87a386e0776
```

**Additional context**
Can provide specific JWT exchange tokens if needed.

**Comment by northtyphoon** [comment]
@weikanglim can you check if you are the classic admin on the subscription? If yes, you can work around it by granting your account the own role on the subscription.

**Comment by weikanglim** [comment]
@northtyphoon Thanks for the suggestion. I did verify indeed that I was a service administrator, likely that this subscription was created awhile back.

![image](https://github.com/Azure/acr/assets/2322434/acbde5ea-af9b-460f-a7b0-5e0fe7b1db29)

And after assigning myself an "Owner" role of the subscription this is addressed.

However, this ends up impacting a lot of users with existing long-lived subscriptions. Would there be an alternative that doesn't require a workaround?

**Comment by northtyphoon** [comment]
@weikanglim we plan to rollout a fix to support the classic administrators by Jan. However, classic administers is deprecated. It will be great to recommend your customer to take the chance to migrate to standard Azure RBAC. 

https://learn.microsoft.com/en-us/azure/role-based-access-control/classic-administrators

```
Important

Classic resources and classic administrators will be [retired on August 31, 2024](https://azure.microsoft.com/updates/cloud-services-retirement-announcement/). Remove unnecessary Co-Administrators and use Azure RBAC for fine-grained access control.
```

**Comment by northtyphoon** [comment]
@shanselman you are right, you still login with MSA. The ramification you can do is granting your MSA account the own role on the subscription. After that, delete the co-admin which will be retried at 8/31/2024.

**Comment by efd7887** [comment]
> @weikanglim can you check if you are the classic admin on the subscription? If yes, you can work around it by granting your account the own role on the subscription.

This is just a terrible idea in practice

**Comment by efd7887** [comment]
Still not fixed.....par for the course with you guys

**Comment by johnsonshi** [comment]
As classic admins have been deprecated, authenticating with them will not be a supported scenario. This will be closed. Feel free to open another ticket if you have issues.
